packages/service-library/src/servicelib/rabbitmq.py

```diff
@@ -131,10 +131,18 @@ async def subscribe(
         message_handler: MessageHandler,
         *,
         exclusive_queue: bool = True,
-    ) -> None:
+        topics: list[str] | None = None,
+    ) -> str:
         """subscribe to exchange_name calling message_handler for every incoming message
         - exclusive_queue: True means that every instance of this application will receive the incoming messages
         - exclusive_queue: False means that only one instance of this application will reveice the incoming message
+
+        specifying a topic will use a TOPIC type of RabbitMQ Exchange instead of FANOUT
+        - topic = "#" will receive every message as # substitutes for 0 or more words
+        - topic = "*.critical" will receive any message with second word set to "critical" (* subsitutes for exactly 1 word)
+        - topic = "red.critical" will receive only messages that are set with key "red.critical"
+
+        NOTE: changing the type of Exchange will create issues if the name is not changed!
```

Comment:
Maybe this should be caught as an error and raised?

Reply:
This actually already raises an error. I am not quite sure yet how to handle that. you cannot change the parameters of an already created exchange. The only way is to delete it and re-create it or to use a different name.

Reply:
I improved the docstring to better explain the case